### DISCUSSION TITLE
What's the difference between TriggerAsync and ResumeAsync

### DISCUSSION THREAD

**Discussion by 463728946** [elsa-workflows/elsa-core#601] [Q&A]
//It doesn't work
            await _invoker.ResumeAsync(instance, new Variables() { ["UserInfo"] = new Variable(userInfo) });

            //It works
            await _invoker.TriggerAsync(nameof(ApproverActivity), new Variables() { ["UserInfo"] = new Variable(userInfo) }, correlationId.ToString());     

           public class ApproverActivity : AuthorizationBaseActivity
    {       
        protected override async Task<ActivityExecutionResult> OnExecuteAsync(WorkflowExecutionContext context, CancellationToken cancellationToken)
        {
            if (!AuthorizationExpression.Unimpeded)            
                return Halt();
            
            var verify = await VerifyAsync(context, cancellationToken);
            if (!verify)           
                return Halt();

            return Done();            
        }

        protected override async Task<bool> OnCanExecuteAsync(WorkflowExecutionContext context, CancellationToken cancellationToken)
        {
            var verify = await VerifyAsync(context, cancellationToken);
            return verify;
        }

        protected override ActivityExecutionResult OnResume(WorkflowExecutionContext context)
        {
            return Outcome(OutcomeNames.Done);           
        }
        
    }

**Comment by 463728946** [comment]
add activityId It's working

await  _invoker.ResumeAsync(instance, new Variables() { ["UserInfo"] = new Variable(userInfo) },instance.BlockingActivities.Select(s=>s.ActivityId));

**Comment by sfmskywalker** [comment]
Glad to see you figure it out :)

To answer your question, the differences between Trigger and Resume are as follows:

* Trigger is a higher-level API that does two things: it starts new workflows and resumes existing workflow instances. It uses Resume internally to do the latter.
* Resume operates on a single, existing workflow instance that is blocked on one or more activities. When invoked, it will resume the workflow, as you've seen.
